fiftyone/core/odm/database.py

```diff
@@ -91,29 +91,29 @@ def get_db_config():
         a :class:`DatabaseConfigDocument`
     """
     conn = get_db_conn()
-    save = False
+
     config_docs = list(conn.config.find())
-    if not config_docs:
-        save = True
-        config = DatabaseConfigDocument(conn)
-    elif len(config_docs) > 1:
-        config = DatabaseConfigDocument(
-            conn, **cleanup_multiple_config_docs(conn, config_docs)
-        )
+    if config_docs:
+        if len(config_docs) > 1:
+            config_doc = _cleanup_multiple_config_docs(conn, config_docs)
+        else:
+            config_doc = config_docs[0]
+
+        config = DatabaseConfigDocument(conn, **config_doc)
+        save = False
     else:
-        config = DatabaseConfigDocument(conn, **config_docs[0])
+        config = DatabaseConfigDocument(conn)
+        save = True
 
     if config.version is None:
         #
-        # The database version was added to this config in v0.15.0, so if no
-        # version is available, assume the database is at the preceding
-        # release. It's okay if the database's version is actually older,
-        # because there are no significant admin migrations prior to v0.14.4.
+        # If the database has no version, then assume the version of the client
+        # that is currently connecting to it.
         #
         # This needs to be implemented here rather than in a migration because
         # this information is required in order to run migrations...
         #
-        config.version = "0.14.4"
+        config.version = foc.VERSION
```

Comment:
Is it possible that someone connects with a greater version and could cause the database version to upgrade and thus brick the connection for everyone else until they upgrade as well? Or require that the datasets be migrated before they can be accessed by everyone else?

Reply:
Yes what you're saying is true and could be bad. However, if the `config` collection is empty, then something has already gone terribly wrong. So the most important contribution of this PR is to (hopefully) remove the possibility that all `config` docs could ever get deleted in the first place.

Using `foc.VERSION` instead of `0.14.4` here has another benefit. Currently, when you connect to a brand new database for the very first time, you see:

```py
import fiftyone as fo; fo.list_datasets()
# Migrating database to v0.24.1
```

because the config version was previously getting set to `0.14.4` and then is immediately migrated up to `foc.VERSION`, which is a pointless migration because, by construction, the database is completely empty. Directly setting the version to `foc.VERSION` prevents the log message, which is cleaner.

Reply:
I kinda mentioned it in the meeting but didn't want to waste everyone's time talking about implementation detail, but I was thinking that instead of automatically choosing the max, it would be safer to set the return_document to before on the update command and after the update check the value of the config version and if the before document's version is greater than the one used for the update command (0.14.4) immediately save the config again using the before docs values. Since the before doc is from the db, even if an error in the local client initially read the value as none, the returned doc will always reflect what was overwritten, so should prevent resetting the value accidentally without simultaneously assuming the connected client has the right version.

Reply:
Alternatively it might just be better to throw an error and force the user to reissue the command as something is wrong with the connection at this point, putting into question whether the results of any subsequent queries in the chain can be accurate

Reply:
I'm not quite understanding your suggestion in the first comment.

Personally I'm still struggling to understand how the `config` collection could become empty in the first place. As explained in the PR description, I could see it happening *IF* the `config` collection contained multiple documents, but... I don't have an explanation for how it could come to contain >1 document, so the mystery still isn't fully solved.

We could start questioning low-level aspects of pymono, like supposing it fails to read document fields correctly and accidentally reads `version=None`, but if that could happen then every aspect of FO could fail, which isn't happening and isn't helpful to think about.

Reply:
To be clear, I don't think there is ever more or less than one config saved to the db. I think it's the one config getting overridden.

Reply:
How do you think the one doc is getting overridden with `version=None`?